src/libslic3r/GCodeWriter.cpp

```diff
@@ -682,6 +682,15 @@ std::string GCodeWriter::extrude_to_xyz(const Vec3d &point, double dE, const std
     return w.string();
 }
 
+std::string GCodeWriter::retract(double retraction_length, bool toolchange)
```

Comment:
Can you explain why you add this function?

Reply:
Because there is no function to retract an arbitrary amount of filament :(

Reply:
Removed it in latest build. Added optional variable to the existing functions.